src/charm.py

```diff
@@ -100,11 +101,11 @@ def __init__(self, *args):
             redis_relation={},
         )
         self._require_nginx_route()
-        self.framework.observe(self.on.start, self._set_up_discourse)
-        self.framework.observe(self.on.upgrade_charm, self._set_up_discourse)
-        self.framework.observe(self.on.discourse_pebble_ready, self._config_changed)
-        self.framework.observe(self.on.config_changed, self._config_changed)
 
+        self.framework.observe(self.on.start, self._on_start)
+        self.framework.observe(self.on.upgrade_charm, self._on_upgrade_charm)
+        self.framework.observe(self.on.discourse_pebble_ready, self._on_discourse_pebble_ready)
+        self.framework.observe(self.on.config_changed, self._config_changed)
```

Comment:
Should the function be called `self._on_config_changed`?

Reply:
Indeed, changed.